torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1949,6 +1949,29 @@ void ONNXShapeTypeInference(
   SetGraphInputTypeReliable(n->owningGraph());
   GRAPH_UPDATE(
       "Running ONNX shape inference for node: ", n->kind().toDisplayString());
+
+  // Go through every output to check if they all have shape
+  // If they all do, this should be reliable even if the Op is not from ONNX.
+  size_t all_output_has_value = 0;
+  for (auto node_output : n->outputs()) {
+    if (auto output_type = node_output->type()->cast<TensorType>()) {
+      if (output_type->dim().has_value()) {
+        // Go through every symbolic_sizes and if one of them > 0, we say this
+        // is set by user. On the other hand, if all of them are *, we take this
+        // node does not have given type, since unreliable nodes have * shape
+        // anyway.
+        for (size_t i = 0; i < output_type->dim().value(); i++) {
```

Comment:
Only check `output_type->dim().has_value()` is not sufficient in graph-level shape type inference, as we already merged tracer shape right after symbolic_fn(the `MergeInferredTypeAndSetMap`), and every node now has dim (even an unreliable node has something like Float(*, *, *, *)). So I set another rule here, if the node has at least one static size (symbolic_sizes()[i].value() < 0 when it's *), it should be seen as type given. Otherwise, it really doesn't matter in terms of the whole dynamic sizes it has.